File: app/src/main/java/com/polok/eubmanagement/presentation/schedule/schedulelist/ScheduleListAdapter.java
package com.polok.eubmanagement.presentation.schedule.schedulelist;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.polok.eubmanagement.databinding.ItemNoticeListBinding;
import com.polok.eubmanagement.model.NoticeData;

import java.util.List;

public class ScheduleListAdapter extends RecyclerView.Adapter<ScheduleListAdapter.NoticeViewHolder> {
    private final List<NoticeData> noticeList;
    private final View.OnClickListener onClickListener;
    private int adapterPosition;
    public ScheduleListAdapter(List<NoticeData> noticeList, View.OnClickListener onClickListener) {
        this.noticeList = noticeList;
        this.onClickListener = onClickListener;
    }

    @NonNull
    @Override
    public NoticeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new NoticeViewHolder(
                ItemNoticeListBinding.inflate(LayoutInflater.from(parent.getContext()),parent,false)
        );
    }

    @Override
    public void onBindViewHolder(@NonNull NoticeViewHolder holder, int position) {
        holder.bind(noticeList.get(position));
    }

    @Override
    public int getItemCount() {
        return noticeList.size();
    }

    public class NoticeViewHolder extends RecyclerView.ViewHolder {
        ItemNoticeListBinding binding;
        public NoticeViewHolder(@NonNull ItemNoticeListBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
        public void bind(NoticeData noticeData) {
            adapterPosition = getAdapterPosition();
            binding.noticeTitle.setText(noticeData.getNotNullText(noticeData.getTitle()));
            binding.noticeDetails.setText(noticeData.getNotNullText(noticeData.getDetails()));
            binding.createdAt.setText(noticeData.getNotNullText(noticeData.getCreatedAt()));

            if (onClickListener != null) {
                itemView.setOnClickListener(v -> {
                    onClickListener.onClick(itemView);
                });
            } else binding.getRoot().setClickable(false);
        }
    }

    public int getAdapterPosition() {return adapterPosition;}
}

File: app/src/main/java/com/polok/eubmanagement/presentation/schedule/schedulelist/ScheduleListAdapter.kt
package com.polok.eubmanagement.presentation.schedule.schedulelist

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.polok.eubmanagement.databinding.ItemScheduleListBinding
import com.polok.eubmanagement.model.ScheduleData
import com.polok.eubmanagement.presentation.schedule.schedulelist.ScheduleListAdapter.ScheduleViewHolder
import com.polok.eubmanagement.util.makeVisible

class ScheduleListAdapter(
    private val onUpdateClickListener: ((ScheduleData?) -> Unit),
) : RecyclerView.Adapter<ScheduleViewHolder>() {
    var isAdmin: Boolean? = null
    private var scheduleList: List<ScheduleData?>? = null

    fun submitList(scheduleList: List<ScheduleData?>?) {
        this.scheduleList = scheduleList
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ScheduleViewHolder {
        return ScheduleViewHolder(
            ItemScheduleListBinding.inflate(
                LayoutInflater.from(parent.context), parent, false
            )
        )
    }

    override fun onBindViewHolder(holder: ScheduleViewHolder, position: Int) {
        holder.bind(scheduleList?.get(position))
    }

    override fun getItemCount(): Int = scheduleList?.size ?: 0

    inner class ScheduleViewHolder(
        private val binding: ItemScheduleListBinding
    ) : RecyclerView.ViewHolder(binding.root) {

        fun bind(scheduleData: ScheduleData?) {
            binding.day.text = String.format("Day\n%s", scheduleData?.day ?: "")
            binding.courseCode.text = String.format("Course Code\n%s", scheduleData?.courseCode ?: "")
            binding.lecturerName.text = String.format("Lecturer Name\n%s", scheduleData?.lecturerName ?: "")
            binding.startEndTime.text = String.format("Start-End Time\n%s", scheduleData?.startEndTime ?: "")
            binding.roomNo.text = String.format("Room No\n%s", scheduleData?.roomNo ?: "")

            if (isAdmin == true) {
                binding.updateDelete.root.makeVisible()
                binding.updateDelete.updateButton.setOnClickListener {
                    onUpdateClickListener(scheduleData)
                }
            }
        }
    }
}
